terminal/tcell/event_test.go

```diff
@@ -112,26 +112,32 @@ func TestToTermdashEvents(t *testing.T) {
 func TestMouseButtons(t *testing.T) {
 	tests := []struct {
 		btnMask tcell.ButtonMask
-		want    mouse.Button
+		want    []mouse.Button
 		wantErr bool
 	}{
-		{btnMask: -1, wantErr: true},
-		{btnMask: tcell.Button1, want: mouse.ButtonLeft},
-		{btnMask: tcell.Button3, want: mouse.ButtonMiddle},
-		{btnMask: tcell.Button2, want: mouse.ButtonRight},
-		{btnMask: tcell.ButtonNone, want: mouse.ButtonRelease},
-		{btnMask: tcell.WheelUp, want: mouse.ButtonWheelUp},
-		{btnMask: tcell.WheelDown, want: mouse.ButtonWheelDown},
+		{btnMask: -1, want: []mouse.Button{mouse.Button(-1)}, wantErr: true},
+		{btnMask: tcell.Button1, want: []mouse.Button{mouse.ButtonLeft}},
+		{btnMask: tcell.Button3, want: []mouse.Button{mouse.ButtonMiddle}},
+		{btnMask: tcell.Button2, want: []mouse.Button{mouse.ButtonRight}},
+		{btnMask: tcell.ButtonNone, want: []mouse.Button{mouse.ButtonRelease}},
+		{btnMask: tcell.WheelUp, want: []mouse.Button{mouse.ButtonWheelUp}},
+		{btnMask: tcell.WheelDown, want: []mouse.Button{mouse.ButtonWheelDown}},
+		{btnMask: tcell.Button1 | tcell.Button2, want: nil},
 	}
 
 	for _, tc := range tests {
 		t.Run(fmt.Sprintf("key:%v want:%v", tc.btnMask, tc.want), func(t *testing.T) {
 
 			evs := toTermdashEvents(tcell.NewEventMouse(0, 0, tc.btnMask, tcell.ModNone))
-			if got, want := len(evs), 1; got != want {
+			if got, want := len(evs), len(tc.want); got != want {
 				t.Fatalf("toTermdashEvents => got %d events, want %d", got, want)
 			}
 
+			// Events that may exist for the terminal implementation but are not valid for termdash will be nil
+			if len(tc.want) == 0 {
```

Comment:
(nit, optional) While the table test structure suggests that test cases here can be written as expecting multiple mouse events (the want value is now a slice), the test logic verifies only the first event in that list (tc.want[0] nelow). That is fine for our usage, but might become confusing in the future.

Can we add a check. so that if future authors specify more than one event in tc.want, they get a readable error message instead of the other events being ignored? We could change this if statement as follows:

```go
switch count := len(tc.want); count {
case 0:
  // Events that may exist for the terminal implementation but are not valid for termdash will be nil
  return
case 1:
  // Proceed with the test.
default:
  t.Fatalf("toTermdashEvents test case specified %d expected events, only one is supported", count)
}
```